onewire/onewire_tinygo.go

```diff
@@ -0,0 +1,29 @@
+package onewire
```

Comment:
You probably wanted to add baremetal build tag for this file to decouple drivers from `machine` package

```suggestion
//go:build baremetal

package onewire
```

Reply:
I've opted for using tinygo build tag instead since the machine.Pin type is tinygo dependent, not baremetal dependent. https://github.com/tinygo-org/tinygo/blob/release/src/machine/machine.go#L42